src/couch_views/src/couch_views_indexer.erl

```diff
@@ -42,6 +42,8 @@ init() ->
         <<"sig">> := JobSig
     } = Data,
 
+    Retries = maps:get(<<"retries">>, Data, 0),
+
```

Comment:
Here we might want to to upgrade the Data map itself, because below in should_retry we expect retries field to be there. Could add a comment about "Upgrade clause, previous versions didn't have a retry field ...".

Reply:
I've changed this and moved it into its own function that upgrades data so we can remove that later if needed.

Reply:
Maybe consider going back to `#{ ... , <<"retries">> := Retries} = Data` since `Data` is not upgraded to always have the <<"retries">> field.